sdk/storage/azure-storage-file-share/src/test/java/com/azure/storage/file/share/FileServiceAsyncAPITests.groovy

```diff
@@ -266,4 +272,69 @@ class FileServiceAsyncAPITests extends APISpec {
         INVALID_ALLOWED_METHOD | 400        | ShareErrorCode.INVALID_XML_NODE_VALUE
 
     }
+
+    def "Restore share min"() {
+        given:
+        def shareClient = primaryFileServiceAsyncClient.getShareAsyncClient(generateShareName())
+        def fileName = generatePathName()
+        def delay = TestMode.PLAYBACK == getTestMode() ? 0L : 30000L
+        def shareItemMono = shareClient.create()
+            .then(shareClient.getFileClient(fileName).create(2))
+            .then(shareClient.delete())
+            .then(Mono.delay(Duration.ofMillis(delay)))
+            .then(primaryFileServiceAsyncClient.listShares(
+                new ListSharesOptions()
+                    .setPrefix(shareClient.getShareName())
+                    .setIncludeDeleted(true)).next())
```

Comment:
I would break this into two parts.

1. Create the resources need using a block (or StepVerifier, either works).
2. Then use sleep if record before going into when.

Right now this feels that setup contains setup and when logic.